### DISCUSSION TITLE
Rename assets folder?

### DISCUSSION THREAD

**Discussion by manuelmeurer** [facebook/docusaurus#8728] [Q&A] [answered]
Is it possible to use something like `/my-assets` instead of `/assets` or is it hardcoded?

**Comment by slorber** [comment, marked as answer]
It is hardcoded. Eventually you can change it by a plugin with the `configureWebpack` lifecycle but it requires modifying out internal config in multiple places (js bundle, CSS bundle, file loader assets...), not so easy.

We have a `OUTPUT_STATIC_ASSETS_DIR_NAME = "assets"` constant, apparently not even used everywhere. We could use an env variable to init it eventually, but I'd prefer not to unless you have a strong use-case to change that name.

**Comment by manuelmeurer** [comment]
My use case is probably not very common - I am rewriting requests in a Cloudflare worker and I need to identify when a request for a Docusaurus asset comes in (/assets is also used by some other requests).
The way I'm solving it right now is `sed`ing through all files after building and replacing `/assets` with `/docusaurus-assets`.

**Reply by slorber** [reply]
I see thanks. Why do you need to identify when a request for a Docusaurus asset comes in? 🤔

**Reply by manuelmeurer** [reply]
Because I want to run Docusaurus on the same domain as our main (Rails) app. 😀

So Docusaurus is running on https://docs.uplink.tech/blog/ but I want to use https://uplink.tech/blog/ instead.
(right now there are a few JS errors since replacing the assets and static folder doesn't work properly yet).
Using a Cloudflare worker, I route all requests to /blog, /knowledge-base and /topics to docs.uplink.tech, and everything else to the main Rails app. Since Rails uses `/assets` as well, it would be the cleanest solution to use `/docs-assets` and `/docs-images` for Docusarus.

**Reply by slorber** [reply]
So both Rails and Docusaurus use `/assets`, that's a conflict, but why do you consider it's Docusaruus fault instead of Rails ? 😅 

My recommendation when interleaving 2 systems in the same domain deployment is to rather always apply a baseurl to one of them to avoid such conflicts. For example, put everything Docusaurus under a `/baseUrl/` path. Here `/assets` is one problem you encountered but in practice this problem can appear on any other static asset that Docusaurus outputs. For example if you had a `docs/assets.md` file maybe we'd have to emit an html file that would lead to Rails problem.

I don't think the goal is to make it possible to configure all possible paths Docusaurus outputs to: you'd rather find a way to keep each deployment sandboxed to a given baseurl instead of trying to merge them.

Note: I don't know nginx, but some hosting providers (and probably self-hosting tools) allow you serve a file-system file **if it exists** and if it doesn't fallback to an alternative (could be the Rails asset route? 🤷‍♂️ )

**Reply by slorber** [reply]
As far as I understand it's possible you change the `/assets` path in Rails so it might solve your problem: https://guides.rubyonrails.org/asset_pipeline.html#changing-the-assets-path

**Reply by manuelmeurer** [reply]
Thanks for your thoughts on this! 🙏 

I definitely don't think it's Docusaurus' fault that their `/assets` clashes with Rails' `/assets`, and I totally understand it's not worth the trouble/complexity to make the asset folder's name configurable.

FWIW, I solved my problem by renaming `/assets` to `/docs-assets` and `/images` to `/docs-images` after building my Docusaurus site, and then fixing the folder names in the generated files with these two regexes:

```ruby
content.gsub! %r(([/"])assets/), "\\1docs-assets/"
content.gsub! %r((?<!docs-assets)/images/), "/docs-images/"
```
